server/aws-lsp-codewhisperer/src/shared/amazonQServiceManager/AmazonQTokenServiceManager.ts

```diff
@@ -318,21 +315,20 @@ export class AmazonQTokenServiceManager extends BaseAmazonQServiceManager<
         }
 
         const parsedArn = parse(newProfileArn)
-        const endpoint = AWS_Q_ENDPOINTS.get(parsedArn.region)
+        const region = parsedArn.region
+        const endpoint = AWS_Q_ENDPOINTS.get(region)
         if (!endpoint) {
             throw new Error('Requested profileArn region is not supported')
         }
 
-        const profiles = await getListAllAvailableProfilesHandler(this.serviceFactory)({
-            connectionType: 'identityCenter',
-            logging: this.logging,
-            token: token,
-            endpoints: new Map([[parsedArn.region, endpoint]]),
-        })
-
-        this.handleTokenCancellationRequest(token)
-
-        const newProfile = profiles.find(el => el.arn === newProfileArn)
+        // Hack to inject a dummy profile name as it's not used by client IDE for now, if client IDE starts consuming name field then we should also pass both profile name and arn from the IDE
+        const newProfile: AmazonQDeveloperProfile = {
+            arn: newProfileArn,
+            name: 'Not being used',
```

Comment:
In case this does get surfaced somewhere unexpectedly, lets make it something less suspicious, like `Client provided profile`

Reply:
makes sense thanks Chris